test-common/src/main/scala/org/scalajs/testcommon/Serializer.scala

```diff
@@ -0,0 +1,290 @@
+package org.scalajs.testcommon
+
+import sbt.testing._
+import java.io._
+
+private[scalajs] trait Serializer[T] {
+  def serialize(x: T, out: Serializer.SerializeState): Unit
+  def deserialize(in: Serializer.DeserializeState): T
+}
+
+private[scalajs] object Serializer extends SerializerPlatformExtensions {
+  /* Serialization states. Serialization is currently stateless, but in the
+   * future we might want to deduplicate things.
+   */
+  final class SerializeState private[Serializer](val s: DataOutputStream)
+      extends AnyVal {
+    def put[T](t: T)(implicit s: Serializer[T]) = s.serialize(t, this)
+  }
+
+  final class DeserializeState private[Serializer](val s: DataInputStream)
+      extends AnyVal {
+    def get[T]()(implicit s: Serializer[T]): T = s.deserialize(this)
+  }
+
+  // Methods to actually perform serialization and deserialization.
+
+  def serialize[T](t: T, out: DataOutputStream)(implicit s: Serializer[T]) = {
+    s.serialize(t, new SerializeState(out))
+  }
+
+  def deserialize[T](in: DataInputStream)(implicit s: Serializer[T]): T = {
+    s.deserialize(new DeserializeState(in))
+  }
+
+  def serialize[T : Serializer](t: T): String =
+    withOutputStream(Serializer.serialize(t, _))
+
+  def deserialize[T : Serializer](s: String): T =
+    withInputStream(s)(Serializer.deserialize[T](_))
+
+  @inline
+  def withInputStream[T](s: String)(body: DataInputStream => T): T = {
+    val in = new DataInputStream(
+        new ByteArrayInputStream(s.getBytes(latin1Charset)))
```

Comment:
Ah, ah, this is how you escape the deprecation of converting Chars to Bytes :) Of course *much* better.